src/utils/daos_hdlr.c

```diff
@@ -1686,99 +1686,113 @@ static int
 dm_serialize_cont_md(struct cmd_args_s *ap, struct dm_args *ca, daos_prop_t *props,
 		     char *preserve_props)
 {
-	int	rc = 0;
+	int	rc = 0, rc2;
 	int	num_attrs = 0;
 	char	**names = NULL;
 	void	**buffers = NULL;
 	size_t	*sizes = NULL;
 	void	*handle = NULL;
-	int (*daos_cont_serialize_md)(char *, daos_prop_t *props, int, char **, char **, size_t *);
+	int	(*serialize_fptr)(char *, daos_prop_t *props, int, char **, char **, size_t *);
+
+	handle = dlopen(LIBSERIALIZE, RTLD_NOW);
+	if (handle == NULL) {
+		DH_PERROR_DER(ap, -DER_INVAL, "libdaos_serialize.so not found");
+		return -DER_INVAL;
+	}
 
 	/* Get all user attributes if any exist */
 	rc = dm_cont_get_usr_attrs(ap, ca->src_coh, &num_attrs, &names, &buffers, &sizes);
 	if (rc != 0) {
 		DH_PERROR_DER(ap, rc, "Failed to get user attributes");
-		D_GOTO(out, rc);
-	}
-	handle = dlopen(LIBSERIALIZE, RTLD_NOW);
-	if (handle == NULL) {
-		rc = -DER_INVAL;
-		DH_PERROR_DER(ap, rc, "libdaos_serialize.so not found");
-		D_GOTO(out, rc);
+		D_GOTO(out_dl, rc);
 	}
-	daos_cont_serialize_md = dlsym(handle, "daos_cont_serialize_md");
-	if (daos_cont_serialize_md == NULL)  {
-		rc = -DER_INVAL;
-		DH_PERROR_DER(ap, rc, "Failed to lookup daos_cont_serialize_md");
-		D_GOTO(out, rc);
+
+	*(void **)(&serialize_fptr) = dlsym(handle, "daos_cont_serialize_md");
+	if (serialize_fptr == NULL)  {
+		DH_PERROR_DER(ap, -DER_INVAL, "Failed to lookup daos_cont_serialize_md");
+		D_GOTO(out_attr, rc = -DER_INVAL);
 	}
-	(*daos_cont_serialize_md)(preserve_props, props, num_attrs, names, (char **)buffers, sizes);
-out:
-	if (num_attrs > 0) {
+	(*serialize_fptr)(preserve_props, props, num_attrs, names, (char **)buffers, sizes);
+out_attr:
+	if (num_attrs > 0)
 		dm_cont_free_usr_attrs(num_attrs, &names, &buffers, &sizes);
-	}
+out_dl:
+	rc2 = dlclose(handle);
+	if (rc2)
+		DH_PERROR_SYS(ap, rc2, "failed to close DL handle");
+	if (rc == 0)
+		rc = rc2;
```

Comment:
Actually this line is wrong as well, this function should return a DER error code so this one what's converting here.